### PR TITLE
Include and library paths must exist

### PR DESCRIPTION
**Pull request by sergiusens**
CXXFLAGS was also missing here.

LP: #1558965

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

File: snapcraft/tests/test_yaml.py [hunk 1]
```diff
@@ -550,12 +557,23 @@ def test_config_snap_environment(self, mock_snapdir):
...
 
-        arch = snapcraft.common.get_arch_triplet()
         for expected in ['foo/lib', 'foo/usr/lib', 'foo/lib/{}'.format(arch),
                          'foo/usr/lib/{}'.format(arch)]:
-            self.assertTrue(expected in paths,
-                            'Expected LD_LIBRARY_PATH to include "{}"'.format(
-                                expected))
+            self.assertTrue(
+                expected in paths,
+                'Expected LD_LIBRARY_PATH in {!r} to include {!r}'.format(
```
Comment (kyrofa) [inline snapcraft/tests/test_yaml.py]: Perhaps `Expected LD_LIBRARY_PATH ({!r}) to include {!r}` would make more sense here?

File: snapcraft/yaml.py [hunk 2]
```diff
@@ -359,6 +342,43 @@ def _build_env(root):
...
+    ]
+
+    library_paths = ['{}{}'.format(prepend, l)
+                     for l in paths if os.path.exists(l)]
+
+    if not library_paths:
+        return ''
+    else:
+        return '{envvar}="{library_paths}{sep}${envvar}"'.format(
+            envvar=envvar, sep=sep, library_paths=sep.join(library_paths))
```
Comment (kyrofa) [inline snapcraft/yaml.py]: Hey, that's pretty slick.

**Comment by kyrofa** [comment]
Looks good to me :+1:

**Comment by sergiusens** [comment]
retest this please

**Comment by elopio** [comment]
retest this please

**Pull request merged**